### ISSUE TITLE
Seed renamed files

### ISSUE THREAD

**Issue by darth-veitcher** [Cloudbox/Cloudbox#351]
**I’m hoping this is something someone else has already solved (or I’m just “doing it wrong”)...**

# Problem
The current setup with `sonarr` and `radarr` renaming files that are downloaded through `rtorrent` results in duplicated files.

as-is process:
* sonarr/radarr send torrent to rtorrent
* rtorrent downloads to `/mnt/local/downloads/torrents/rutorrent/completed/`
* on completion Sonarr/radarr `copy` files (whilst renaming them) across to `/mnt/local/Media/{Movies,TV}`
* when storage in `/mnt/local/Media` reaches a certain utilisation `cloudplow` then moves all files across to `/mnt/remote/Media`.

What therefore happens is that local storage will always run out as `/mnt/local/downloads` is never cleared down despite files being available under `/mnt/unionfs/Media/{Movies,TV}` in their renamed state.

# Solution
I would like the process to, instead, look similar to the below:
* sonarr/radarr send torrent to rtorrent
* rtorrent downloads to `/mnt/local/downloads/torrents/rutorrent/completed/`
* on completion Sonarr/radarr `copy` files (whilst renaming them) across to `/mnt/local/Media/{Movies,TV}`
* **rtorrent stopped for torrent and repointed to renamed files in `/mnt/unionfs/Media`**
* **rtorrent restarted, seeding from `unionfs`**
* **original files in `/mnt/local/downloads` deleted**
* when storage in `/mnt/local/Media` reaches a certain utilisation `cloudplow` then moves all files across to `/mnt/remote/Media`.

# Additional context
As it looks like rtorrent doesn't support the renaming of files (just the root folder) it's probably something we’d need to use a client like Deluge for. 

The [API docs](https://deluge.readthedocs.io/en/latest/reference/api.html) for `rename_files` and `rename_folder` look useful.

Potentially most graceful way to do this is insert a new container for `deluge` just to cater for longer-term seeding of files. That way it can act as a drop-in without interrupting standard workflow (unless people want to default to deluge for everything as an install option). 

Sonarr also has a [documented API](https://github.com/Sonarr/Sonarr/wiki/API) as well as a [database](https://github.com/Sonarr/Sonarr/wiki/AppData-Directory) you can direct access which might help if it’s something we need to build a script for ourselves.

**Comment by kultigspritzig** [comment]
1. First of all, files shouldn't be copied from /mnt/local/downloads, they should be hard linked to avoid unnecessary disk operations and space consumption. Radarr supports that, but it isn't a default behavior recommended by the wiki (_Use Hardlinks instead of Copy: No_) for some reason.
2. Once the file is uploaded and ready to be plowed, the torrent should be paused and the /mnt/local/downloads hardlink replaced with a symlink pointing to /mnt/unionfs/ copy of the files. Then after resuming everything will be transparent to ruTorrent and there's no need to rename anything in the client itself.

**Comment by saltydk** [comment]
Seeding from /mnt/unionfs/Media will never be the standard use case for cloudbox.

**Comment by kultigspritzig** [comment]
> Seeding from /mnt/unionfs/Media will never be the standard use case for cloudbox.

True, but just being able to have that option can help build a good reputation with private trackers, even with reduced upload speeds.

**Comment by saltydk** [comment]
My reason for stating that is that it is already an option if you customize Cloudbox on your own, with very minor changes, so there is no reason for us to add an option. As with many other advanced use cases it requires some know how to make it work and thus not wise to integrate as the end result is just more questions on discord about why it isn't working when they change the default settings. Just my POV.

**Comment by kultigspritzig** [comment]
@saltydk can you elaborate a bit? I analyzed all the solutions possible, mergerfs, unionfs, downloading directly to union/mergerfs, sonarr/radarr mapping, but still nothing can really work without using symlinking

**Comment by saltydk** [comment]
First of you need to let go of wanting to link Media and downloads folders together since that creates an unnecessary line of problems for you to solve. You can just have cloudplow upload whatever folder you are seeding from and have that folder be used under the /mnt/unionfs mergerfs structure. 

Personally, I don't do seeding from rclone gdrive remote since I have not found a way to get around massive amount of IOWait which cripples performance. Secondly you'd need to restart the torrent client regularly if you're not using a script to move from one client to another otherwise mergerfs will prevent file deletions (files uploaded to cloudplow) due to the torrent client still having the files open. 

I know some people have scripts that move torrents from deluge to rtorrent somewhere since deluge is preferred for initial swams (racing) and rtorrent for long term. So that could be one way to handle the file locks, mileage may vary.

TLDR; Seeding from rclone remote is not without problems.

**Comment by kultigspritzig** [comment]
Thanks. After thinking about it a bit more, given the random I/O access requirement of torrent clients and the resulting IOWaits, this is indeed not the best idea.
